conformance/utils/http/http.go

```diff
@@ -253,14 +272,37 @@ func WaitForConsistentResponse(t *testing.T, r roundtripper.RoundTripper, req ro
 	tlog.Logf(t, "Request passed")
 }
 
-func CompareRequest(t *testing.T, req *roundtripper.Request, cReq *roundtripper.CapturedRequest, cRes *roundtripper.CapturedResponse, expected ExpectedResponse) error {
+// WaitForConsistentFailureResponse repeats the provided request for some period
+// of time checking thatthe error was returned. This function fails when http
+// Status OK (200) was returned.
+func WaitForConsistentFailureResponse(t *testing.T, r roundtripper.RoundTripper, req roundtripper.Request, threshold int, maxTimeToConsistency time.Duration) {
+	AwaitConvergence(t, threshold, maxTimeToConsistency, func(elapsed time.Duration) bool {
+		_, cRes, err := r.CaptureRoundTrip(req)
+		if err != nil {
+			tlog.Logf(t, "Request failed, not ready yet: %v (after %v)", err.Error(), elapsed)
+			return false
+		}
+		if roundtripper.IsTimeoutError(cRes.StatusCode) {
+			tlog.Logf(t, "Response expectation failed for request: %+v  not ready yet: %v (after %v)", req, cRes.StatusCode, elapsed)
+			return false
+		}
+		if cRes.StatusCode == 200 { // http:StatusCode OK
+			t.Fatalf("Request %+v should failed, returned HTTP Status OK (200) instead", req)
+			return false
+		}
```

Comment:
Using `t.Fatalf` doesn't seem like the right idiom for AwaitConvergence.  It would just break the loop, wouldn't it?  Maybe something like this is fine:
```suggestion
		if roundtripper.IsTimeoutError(cRes.StatusCode) || cRes.StatusCode == 200 {
			tlog.Logf(t, "expected status code to be 200, got %d for request: %+v  (after %v)", cRes.StatusCode, req, elapsed)
			return false
		}
```